plugins/imgur.py

```diff
@@ -128,7 +129,15 @@ def export_submission(self,
             if is_album:
                 results['link_display'] = '[Imgur Album](https://imgur.com/a/%s)  \n' % album['id']
             else:
-                results['link_display'] = '[Imgur](%s)  \n' % images[0]['link'].replace('http', 'https')
+                
+                picture_url = images[0]['link'].replace('http', 'https')
+                r = requests.head(picture_url)
+                mime_text = r.headers.get('Content-Type')
+                mime = mimeparse.parse_mime_type(mime_text)
+                if mime[1] == 'gif':
+                    picture_url = re.sub('\.gif|jpg|png', '.gifv', picture_url)
```

Comment:
Hrm, looking at it a little more closely, it may be safer to use a regex than a split, but it wouldn't be this one exactly. The way it's constructed right now, this could happen:

```python
>>> picture_url = 'https://i.imgur.com/ijpg9SV.gif'
>>> re.sub(r'\.gif|jpg|png', '.gifv', picture_url)
'https://i.imgur.com/i.gifv9SV.gifv'
```

Which is obviously not what is wanted.
I'd probably do something more like this:

```python
>>> re.sub(r'(\.\w+)?$', '.gifv', 'https://i.imgur.com/ijpg9SV.gif')
'https://i.imgur.com/ijpg9SV.gifv'
>>> re.sub(r'(\.\w+)?$', '.gifv', 'https://i.imgur.com/ijpg9SV.jpg')
'https://i.imgur.com/ijpg9SV.gifv'
>>> re.sub(r'(\.\w+)?$', '.gifv', 'https://i.imgur.com/ijpg9SV')
'https://i.imgur.com/ijpg9SV.gifv'
```

I use the `r'text \ here'` syntax which allows me to insert backslashes in a string without having to escape them. Breaking down the regular expression `(\.\w+)?$`, it's:

- `(...)?$`: Optionally select a group at the end of the string
- `\.`: Match the literal dot character (I can use one `\` because I used `r''` for the whole string)
- `\w`: Match a word character (alphabetical, number, or underscore)
- `+`: Match one or more of the previous expression (in this case, `\w`)

So, all together, it's "If it exists, select the bit at the end starting with a dot and a bunch of word characters. Otherwise, just select the end. Then, replace the selection with `.gifv`."

Take a look at the [re](https://docs.python.org/3/library/re.html) module for more info.